lib/grp.gi

```diff
@@ -329,7 +333,7 @@ local gen, s;
       break;
     fi;
   od;
-  return Factors( s )[1];
+  return SmallestRootInt( s );
```

Comment:
Yes, this must be superior to the previous code. I wonder how I did not think of it when I checked it a couple years ago.....